python/tvm/driver/tvmc/composite_target.py

```diff
@@ -22,7 +22,7 @@
 from tvm.relay.op.contrib.arm_compute_lib import partition_for_arm_compute_lib
 from tvm.relay.op.contrib.ethosn import partition_for_ethosn
 from tvm.relay.op.contrib.vitis_ai import partition_for_vitis_ai
-from tvm.contrib.target import vitis_ai
+from tvm.contrib.target import vitis_ai  # pylint: disable=unused-import
```

Comment:
Out of curiosity: why do we need this import here? is this for some module initialisation purposes?

Reply:
The `relay.ext.vitis_ai` codegen is registered there and won't be found if we don't import the module: https://github.com/apache/tvm/blob/5d5bbfb1ed2281c3e9b2719be6643281d60022fb/python/tvm/contrib/target/vitis_ai.py#L64

Reply:
Understood. So that makes `pyxir` a mandatory dependency for TVMC, which is not a problem. You just need to declare it here:
https://github.com/apache/tvm/blob/760e9b2b0423f5b66f206d9a9195a1b2b5a007a3/python/gen_requirements.py#L128-L141

Reply:
Actually I was thinking how could we avoid adding frontend/codegen specific packages to be a TVMC mandatory dependency. For example, it's weird if a user doesn't enable Xilinx flow in `config.cmake` but is asked to install `pyxir` to run TVMC.

Maybe we should declare partition functions at one place and import related dependencies lazily.

Reply:
Yes, indeed, I agree that we should try to avoid having unnecessary dependencies.  I was thinking about allowing to add setup functions in which we can lazily load dependencies (or do other stuff) to the registered codegens: https://github.com/apache/tvm/blob/5cf67bc00c31069d998ec85d4786a8de830ea597/python/tvm/driver/tvmc/composite_target.py#L36

Reply:
Agreed. This is maybe another callback we can register, and call within `get_codegen_by_target`?

Reply:
Yes, I am thinking of adding `init` key with corresponding initialization function like for example:
```
"vitis-ai": {
        "config_key": "relay.ext.vitis_ai.options",
        "init": init_for_vitis_ai,
        "pass_pipeline": partition_for_vitis_ai,
    },
```
I will work on something and add it to the PR. Are you ok with "init" or do you prefer some other name?

Reply:
I agree with `init` naming.